java/arcs/sdk/jvm/CollectionImpl.kt

```diff
@@ -11,35 +11,39 @@
 
 package arcs.sdk
 
+import arcs.core.common.Referencable
+import arcs.core.crdt.CrdtSet
+
 /** [ReadWriteCollection] implementation for the JVM. */
-@Suppress("UNUSED_PARAMETER")
-// TODO: Connect to storage.
-class CollectionImpl<T : Entity>(
-    val particle: Particle,
+class CollectionImpl<T>(
     override val name: String,
-    entitySpec: EntitySpec<T>
-) : ReadWriteCollection<T> {
-    private val entities = mutableListOf<T>()
-
+    particle: Particle,
+    storageProxy: StorageProxy<CrdtSet.Data<T>, CrdtSet.IOperation<T>, Set<T>>
+) : HandleImpl<CrdtSet.Data<T>, CrdtSet.IOperation<T>, Set<T>>(name, particle, storageProxy),
+    ReadWriteCollection<T>
+    where T : Entity, T : Referencable {
```

Comment:
Is `Entity` not `Referencable` already? That seems like it's something which could/should be resolved.

Reply:
I agree. Entity should just extend Referencable. That could be in a follow up PR, but please add a TODO :-)

Reply:
I had avoided it because of dependency issues in wasm. I've made a stab at resolving them in this PR, since I needed to fix it, anyway.